internal/pkg/service/cli/options/options.go

```diff
@@ -43,6 +44,7 @@ func New() *Options {
 		envNaming: envNaming,
 		setBy:     make(map[string]cliconfig.SetBy),
 		parser:    viper.New(),
+		LogFormat: "console",
```

Comment:
Valid default values are always good practice.

Did it fall in NewLogFormat? (there will be no problem with the string type).

----

It the switch bellow, you preffered default encoder over error. It is IMHO ok, for style flags like log format.
So now there is no place for it to fall?

![image](https://github.com/keboola/keboola-as-code/assets/19371734/1b9add3d-53fb-4f3b-8e77-3abfcd592834)

Reply:
> Did it fall in NewLogFormat? (there will be no problem with the string type).

Yes it did because without the default set here it got empty string as input. I don't see why this would change with string type. Inputs other than "json" and "console" are still invalid.

Reply:
> So now there is no place for it to fall?

Yeah the default branch is technically unnecessary. Should I remove it? I'm unsure what the best practice is around enums and switches in Go. In PHP we have static analysis that checks that every such switch uses all the possible values of the enum and has no default branch but I don't think such advanced rule exists for Go linter.

Reply:
> but I don't think such advanced rule exists for Go linter.
No, there is no such linter.

Default branch should always be there, you have 3 options:
- panic(errors.New("this is a programmer error"))
- return errors.New("this error can also occur during normal operation (for example user input)")
- default behaviour - I don't use this almost at all, but with log-format it makes sense